logs/querysample/tracing_test.go

```diff
@@ -0,0 +1,83 @@
+package querysample
+
+import (
+	"testing"
+	"time"
+
+	"github.com/pganalyze/collector/state"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type startAndEndTimeTestPair struct {
+	testName   string
+	traceState trace.TraceState
+	sample     state.PostgresQuerySample
+	startTime  time.Time
+	endTime    time.Time
+}
+
+func TestStartAndEndTime(t *testing.T) {
+	currentTime := time.Date(2023, time.January, 1, 1, 2, 3, 456*1000*1000, time.UTC)
+	traceState := trace.TraceState{}
+	otelTraceState, err := traceState.Insert("ot", "p:8;r:62")
+	if err != nil {
+		t.Fatalf("Failed to initialize object: %v", err)
+	}
+	pganalyzeTraceStateWithoutT, err := traceState.Insert("pganalyze", "x:foo;y:bar")
+	if err != nil {
+		t.Fatalf("Failed to initialize object: %v", err)
+	}
+	// inserting the same key will update the value
+	pganalyzeTraceState, err := traceState.Insert("pganalyze", "t:1697666938.6297212")
```

Comment:
Does this mutate `traceState`? If so, it might be nice to just initialize a new one to better isolate the individual tests.

Reply:
ah I was actually wrong, this is returning the copied tracestate 🤦 will update the comment, also will tweak things a bit